bson/_cbsonmodule.c

```diff
@@ -447,6 +447,30 @@ static long _type_marker(PyObject* object) {
     return type;
 }
 
+/* Fill out a type_registry_t* from a TypeRegistry object.
+ *
+ * Return 1 on success. options->document_class is a new reference.
+ * Return 0 on failure.
+ */
+int convert_type_registry(PyObject* registry_obj, type_registry_t* registry) {
+    registry->encoder_map = PyObject_GetAttrString(registry_obj, "_encoder_map");
```

Comment:
Need to initialize all the `registry` fields to NULL up top. Technically we only need to initialize `registry->decoder_map` but it's good practice if we do it for all the fields.